### ISSUE TITLE
[@typescript-eslint/no-unused-vars] false positive in type declarations

### ISSUE THREAD

**Issue by erwanvivien** [typescript-eslint/typescript-eslint#4336]
False positive for this piece of code.

Coming from [StackOverflow](https://stackoverflow.com/questions/63767199/typescript-eslint-no-unused-vars-false-positive-in-type-declarations)

```tsx
type ButtonProps = {
  onPress?: (event: GestureResponderEvent) => void;
}

// 'event' is defined but never used. eslint(no-unused-vars)

/// Also:

Button.defaultProps = {
  image: { uri: null, imagePosition: 'left' },
  style: {},
  onPress: (event: GestureResponderEvent) => { },
};

// 'event' is defined but never used. eslint(no-unused-vars)
// 'event' is declared but its value is never read. ts(6133)
```

My package.json eslint related versions
```
    "@typescript-eslint/eslint-plugin": "^5.8.0",
    "@typescript-eslint/parser": "^5.8.0",
    "eslint": "^8.5.0",
    "eslint-config-airbnb": "^19.0.2",
    "eslint-plugin-import": "^2.25.3",
    "eslint-plugin-jsx-a11y": "^6.5.1",
    "eslint-plugin-react": "^7.27.1",
    "eslint-plugin-react-hooks": "^4.3.0",
    "typescript": "~4.3.5"
```

**Comment by Josh-Cena** [comment]
Same as #4337, you have to disable `no-unused-vars`: https://typescript-eslint.io/rules/no-unused-vars#how-to-use

**Comment by bradzacher** [comment]
As with the other issue. Please don't ignore the issue template. It exists for good reason.
